src/main/scala/com.snowplowanalytics.sauna/apis/UrbanAirship.scala

```diff
@@ -0,0 +1,166 @@
+package com.snowplowanalytics.sauna
+package apis
+
+
+
+import java.io.DataOutputStream
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext
+import scala.util.{ Failure, Success }
+import org.apache.commons.io.IOUtils
+import java.io.File
+import java.util.Properties
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.io.InputStream
+import java.util.ArrayList
+import java.io.InputStream
+import java.util.HashMap
+import java.util.Iterator
+import scala.io.Source.fromFile
+import java.net.URL
+import akka.actor.{ActorRef, Props}
+import java.util.Map.Entry
+import scala.io.Source
+import scala.collection
+import scala.collection.mutable.{ Map => MutableMap }
+import javax.net.ssl.HttpsURLConnection
+import java.util.zip.GZIPOutputStream
+import org.json.JSONObject
+import responders.urbanairship.UAResponder.Airship
+import scala.collection.immutable.ListMap
+import play.api.libs.json.Json
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class UrbanAirship(implicit logger: ActorRef) {
+  import UrbanAirship._
+  
+//  implicit val ec = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(10))
+  def MaptoRequest(appMap: Map[String, Map[String, List[Airship]]]): Unit =
```

Comment:
1. It's a common practice to call methods in camelCase, not in PascalCase.
2. There's a nice feature in Scala called type alias. Basically, using type aliases you can shorten some not very beautiful and self-describing type like `Map[String, List[Airship]]` into `IdentifierMap`. This is optional, but neat.
3. Long methods with `Unit` is what we want to avoid. They're untestable and unpredictable. It's fine to have them, but it should be composed from many small "pure" functions. I'll post details in comments below.